### PR TITLE
Add besselj0 tests spanning the full gamut of magnitudes

### PR DESCRIPTION
**Pull request by rreusser**
## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds a test for the numbers `1e-323, 1e-322, ..., 1e306, 1e307, 1e308` as well as for `-1e-323, -1e-322, ..., -1e306, -1e307, -1e308`. There are a number of cutoffs that switch to different evaluation strategies, so although test coverage would ensure all branches are exercised, I thought it was worthwhile to just plug in numbers of every possible order of magnitude to increase confidence that there aren't major gaps in the testing, which is not otherwise readily apparent. A tolerance of `16ϵ` against julia's output appears to satisfy all numbers tested.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

**Review by Planeshifter — APPROVED** [review]
LGTM.

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
Once CircleCI passes, @Planeshifter can merge.

File: lib/node_modules/@stdlib/math/base/special/besselj0/test/test.js [hunk 1]
```diff
@@ -240,6 +242,52 @@ tape( 'the function evaluates the Bessel function of the first kind of zero orde
 	t.end();
 });
 
+tape( 'the function evaluates the Bessel function of the first kind of order zero (J_0) or positive x over a wide range of magnitudes', function test( t ) {
```
Comment (rreusser) [inline lib/node_modules/@stdlib/math/base/special/besselj0/test/test.js]: silly typo. one sec.

**Review by kgryte — APPROVED** [review]
Thanks for catching that!

File: lib/node_modules/@stdlib/math/base/special/besselj0/test/test.js (hunk 1 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/besselj0/test/test.js]: Fixed.

**Pull request merged**